### PR TITLE
Upg: execute lint in pre-commit hook if DUST_LINT_ON_COMMIT=1

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Lint on pre-commit if DUST_LINT_ON_COMMIT=1.

## Risk

None

## Deploy Plan

None

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -10,6 +10,29 @@ if [ "$current_branch" = "main" ]; then
     fi
 fi
 
-# Allow commit
-exit 0
+# If env var viDUST_LINT_ON_COMMIT = 1, run the linter
```
Comment (flvndvd) [inline .husky/pre-commit:13]:
```suggestion
# If env var DUST_LINT_ON_COMMIT = 1, run the linter
```

File: .husky/pre-commit [hunk 2]
```diff
@@ -10,6 +10,29 @@ if [ "$current_branch" = "main" ]; then
...
+root_path=$(git rev-parse --show-toplevel)
 
+# List the directories to check for changes
+directories=("front" "connectors")
+
+# For each directory, check if any files have changed
+for directory in "${directories[@]}"; do
+    if git diff --cached --name-only | grep -qE "^$directory/"; then
+        # Run the linter on root_path concatenated with the directory
+        if ! npm run lint --prefix $root_path/$directory; then
```
Comment (flvndvd) [inline .husky/pre-commit:29]: Usually people go with https://www.npmjs.com/package/lint-staged that pretty output linter errors.

**Review by flvndvd — APPROVED** [review]

File: .husky/pre-commit (hunk 1 above)
Comment (Fraggle) [reply .husky/pre-commit]: OOps :)

File: .husky/pre-commit (hunk 2 above)
Comment (Fraggle) [reply .husky/pre-commit:29]: Thanks for the pointer, i'll keep the PR as it to avoid distracting myself too much :)

**Pull request merged**